src/toplevel.ml

```diff
@@ -0,0 +1,158 @@
+open Stdune
+
+let toplevel_dir_prefix = ".toplevel."
+
+module Source = struct
+  type t =
+    { name : string
+    ; dir : Path.t
+    ; loc : Loc.t
+    ; main : string
+    }
+
+  let main_module_name t = Module.Name.of_string t.name
+  let main_module_filename t = t.name ^ ".ml"
+  let source_path t = Path.relative t.dir (main_module_filename t)
+
+  let obj_dir { dir; name ; _ } =
+    Obj_dir.make_exe ~dir ~name
+
+  let exe t =
+    (* Use the [.exe] version. As the utop executable is declared with [(modes
```

Comment:
There are a few mention to utop in this file that should probably be replaced by something more generic.

Reply:
Done. This function wasn't used so I just deleted it.